tests/test_utils.py

```diff
@@ -300,6 +301,24 @@ def test_url_matches(pattern, url, expected):
     assert pattern.matches(httpx.URL(url)) == expected
 
 
+@pytest.mark.parametrize(
+    ["value", "expected"],
+    [
+        (b"value", b"value"),
+        (b"success", b"success"),
+    ],
+)
+def test_normalize_header_value(value, expected):
+    assert normalize_header_value(value) == expected
+
+
+def test_normalize_header_incorrect_value():
+    with pytest.raises(
+        TypeError, match=f"Header value must be str or bytes, not {type(None)}"
+    ):
+        normalize_header_value(None)  # type: ignore
```

Comment:
Could we have these tests in `tests/client/test_headers.py` with the testing against `httpx.Header(...)` instead of against the internal API?